lib/jelix-modules/jacl2db/classes/jAcl2DbManager.class.php

```diff
@@ -127,16 +128,37 @@ public static function setRightsOnGroup($group, $rights)
             $oldrights[$rec->id_aclsbj] = ($rec->canceled ? 'n' : 'y');
         }
 
+        $roots = array();
+        foreach ($subjects as $subject) {
+            $matches = array();
+            if (preg_match('/(.*)(\.view)$/', $subject->id_aclsbj, $matches)) {
+                $roots[] = $matches[1];
+            }
+        }
+
         // set new rights.  we modify $oldrights in order to have
         // only deprecated rights in $oldrights
         foreach ($rights as $sbj => $val) {
             if ($val === '' || $val == false) {
                 // remove
             } elseif ($val === true || $val == 'y') {
+                foreach ($roots as $root) {
+                    if (preg_match('/^('.$root.'.)/', $sbj)) {
```

Comment:
using a variable as is in a regexp could generate a bad regexp. For example here, the variable can contain some characters like  `.` that are regexp operator. Except if the string is really a part of regexp, you should use `preg_quote`.

Anyway, probably here it is better to use `strpos(...) === 0` instead of a regexp, more efficient.